app.py

```
@@ -1,4 +1,75 @@
+                # No data found - render template with no_data=True
+                return render_template('index.html', 
+                                       selected_date=selected_date,
+                                       no_data=True,
+                                       courses=courses)
+            else:
+                # Data found - render with attendance_data
+                return render_template('index.html', 
+                                       selected_date=selected_date,
+                                       attendance_data=attendance_data,
+                                       courses=courses)
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)